ts/features/bonus/siciliaVola/screens/voucherList/VoucherDetailsScreen.tsx

```diff
@@ -0,0 +1,202 @@
+import {
+  selectedVoucherCodeSelector,
+  selectedVoucherSelector
+} from "../../store/reducers/selectedVoucher";
+import { H3 } from "../../../../../components/core/typography/H3";
+import CopyButtonComponent from "../../../../../components/CopyButtonComponent";
+import { isLoading, isReady } from "../../../bpd/model/RemoteValue";
+import { H4 } from "../../../../../components/core/typography/H4";
+import { formatDateAsLocal } from "../../../../../utils/dates";
+import { useIOBottomSheetRaw } from "../../../../../utils/bottomSheet";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { LoadingErrorComponent } from "../../../bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import VoucherDetailBottomSheet from "../../components/VoucherDetailBottomsheet";
+import { fromVoucherToDestinationLabels } from "../../utils";
+import { navigateToWorkunitGenericFailureScreen } from "../../../../../store/actions/navigation";
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+const styles = StyleSheet.create({
+  itemRow: {
+    flex: 1,
+    flexDirection: "row",
+    justifyContent: "space-between"
+  }
+});
+
+const VoucherDetailsScreen = (props: Props): React.ReactElement | null => {
+  useEffect(() => {
+    if (props.selectedVoucherCode !== undefined) {
+      props.getVoucherDetail(props.selectedVoucherCode);
+    }
+  }, []);
+  const { present, dismiss } = useIOBottomSheetRaw(650);
+
+  // The selectedVoucherCode can't be undefined in this screen
+  if (!isReady(props.selectedVoucher)) {
+    return fromNullable(props.selectedVoucherCode).fold(null, svc => (
+      <BaseScreenComponent
+        goBack={true}
+        contextualHelp={emptyContextualHelp}
+        headerTitle={I18n.t("bonus.sv.headerTitle")}
+      >
+        <LoadingErrorComponent
+          isLoading={isLoading(props.selectedVoucher)}
+          loadingCaption={I18n.t("global.remoteStates.loading")}
+          onRetry={() => props.getVoucherDetail(svc)}
+        />
+      </BaseScreenComponent>
+    ));
+  }
+
+  const selectedVoucher = props.selectedVoucher.value;
+  const openBottomSheet = () =>
+    present(
+      <VoucherDetailBottomSheet
+        barCode={selectedVoucher.barCode}
+        qrCode={selectedVoucher.qrCode}
+        onExit={() => dismiss()}
+      />,
+      I18n.t("bonus.sv.components.voucherBottomsheet.title")
+    );
+  const voucherRevocationButtonProps = {
+    bordered: true,
+    style: {
+      borderColor: IOColors.red
+    },
+    labelColor: IOColors.red,
+    onPress: props.back,
+    title: I18n.t("bonus.sv.voucherList.details.cta.voucherRevocation")
+  };
+  const openQrButtonProps = {
+    primary: true,
+    bordered: false,
+    onPress: () => openBottomSheet(),
+    title: I18n.t("bonus.sv.voucherList.details.cta.openQr")
+  };
+
+  const voucherId = selectedVoucher.id ? selectedVoucher.id.toString() : "";
```

Comment:
```suggestion
  const voucherId = selectedVoucher.id?.toString() ?? "";
```
some but shorter

Reply:
Updated in [aff981c](https://github.com/pagopa/io-app/pull/3400/commits/aff981c96caa17dc974f5fabcbf0b37806ea5b78)